Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,606 @@
+        /// Gets or sets the font style.
+        /// </summary>
+        public FontStyle FontStyle
+        {
+            get
+            {
+                return _fontStyle ?? FontStyle.Normal;
+            }
+
+            set
+            {
+                if (_fontStyle != value)
+                {
+                    _fontStyle = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_fontStyleName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the font weight or thickness.
+        /// </summary>
+        public FontWeight FontWeight
+        {
+            get
+            {
+                return _fontWeight ?? FontWeights.Normal;
+            }
+
+            set
+            {
+                if (!_fontWeight.HasValue || _fontWeight.Value.Weight != value.Weight)
+                {
+                    _fontWeight = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_fontWeightName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a brush that describes the foreground of the column cells.
+        /// </summary>
+        public Brush Foreground
+        {
+            get
+            {
+                return _foreground;
+            }
+
+            set
+            {
+                if (_foreground != value)
+                {
+                    _foreground = value;
+                    NotifyPropertyChanged(DATAGRIDCOMBOBOXCOLUMN_foregroundName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets a <see cref="T:Windows.UI.Xaml.Controls.ComboBox"/> control that is bound to the column's ItemsSource collection.
+        /// </summary>
+        /// <param name="cell">The cell that will contain the generated element.</param>
+        /// <param name="dataItem">The data item represented by the row that contains the intended cell.</param>
+        /// <returns>A new <see cref="T:Windows.UI.Xaml.Controls.ComboBox"/> control that is bound to the column's ItemsSource collection.</returns>
+        protected override FrameworkElement GenerateEditingElement(DataGridCell cell, object dataItem)
+        {
+            EnsureColumnBinding(dataItem);
+
+            EnsureDisplayMemberPathExists();
+
+            var comboBox = new ComboBox
+            {
+                Margin = default(Thickness),
+                HorizontalAlignment = HorizontalAlignment.Stretch,
+                VerticalAlignment = VerticalAlignment.Center
+            };
+
+            if (dataItem != null)
+            {
+                var value = dataItem.GetType().GetProperty(Binding.Path.Path).GetValue(dataItem);
+
+                var items = ItemsSource as IEnumerable<object>;
+
+                var selection = !string.IsNullOrEmpty(DisplayMemberPath)
+                    ? items?.FirstOrDefault(x => x.GetType().GetProperty(Binding.Path.Path).GetValue(x).Equals(value))
```

Comment:
Hello Filip, I did some more testing with DisplayMemberPath and found a few issues.

This is what I did:
DataGrid is bound to a list of Person objects, where Person has a property called ZipCode.
DataGridComboBoxColumn.ItemsSource is a list of Address objects, where Address has a property called Zip.
DataGridComboBoxColumn.DisplayMemberPath is set to "Zip".
DataGridComboBoxColumn.Binding.Path is set to "ZipCode".

So this line should be using this.DisplayMemberPath instead of Binding.Path.Path.

Reply:
Thank you for the comment. I actually don't think the code is incorrect here instead I believe you have misunderstood how I intended for it to be used. Let me explain:

DisplayMemberPath is only the property shown in ComboBox drop down list and can be any property within the ComboBox ItemsSource element. However, the ItemsSource elements of both the DataGrid and the ComboBoxColumn needs to have a property with the same name that Binding.Path.Path can point to.

In your example, the Address object's Zip property would have to be renamed to ZipCode as the name must match Binding.Path.Path. If the Address object contains another property called Street, you could set DisplayMemberPath="Street" to make the Street name appear in the both the TextBlock and the ComboBox of the ComboBoxColumn. Once a Street is selected, that street's ZipCode would be placed in the Person object's ZipCode property.

If we want to be able to connect to a property with an arbitrary name, e.g. Zip, we would need to introduce one more path property in the ComboBoxColumn. We could call it something like SetBindPath. What do you think?

Reply:
@MSRegisB I fail to see the difference between your two lists. Also, Persons should be bound to the DataGrid's ItemsSource right? Not the Columns? The list is then bound to the Column's ItemsSource to provide a selection of ZipCodes.

I used object as it seems to be the convention for other controls with an ItemsSource like the ListView and GridView. I then cast it to IEnumerable<object> not IEnumerable, but I would think it would be possible to cast to both of these. 

I don't understand why you cannot set ItemsSource to a List<int>. Are you trying to set the ItemsSource to an int directly? Can you post or send me your code so I can have a look please?

Reply:
@MSRegisB After some testing I now understand the problem. Casting to `IEnumerable<object>` wont work with non-nullable types.

I guess we could change either the declaration to IEnumerable or just the casting? What would you prefer?

Reply:
Wow, this is strange. My previous comment does not show what I actually typed. When I hit its ... menu and then its Edit command, it shows my actual text. Somehow it's hiding the < int >  and < object > parts by default.

Anyways, I'm glad you could figure out the problem I hit. I think it would be good to support both List < int > and List < object > for the combobox's ItemsSource. 

If you can figure out how to do that, I'd change the ItemsSource type from object to IEnumerable.
If you cannot figure out how to do that, I'd change the ItemsSource type from object to IEnumerable<object>.

Hope this makes sense.  Thank you.

Reply:
@filipwa84   Cool - it worked for me.  I would suggest one last change:  
Replace the DataGridComboBoxColumn.ItemsSource type from object to IEnumerable. And then in the code you can get rid of all the 'var items = ...' variables and use this.ItemsSource instead.   Hope this makes sense.   Thanks!